### PR TITLE
Add compaction for topic

### PR DESCRIPTION
**Pull request by abramin**

### CODE REVIEW

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: dev-aws
Module: account-platform/kafka-shared
Path: dev-aws/account-platform/
Commit ID: c3873fb948f7407a9d44f5394fa6dbf2de9b5160
Requested At: 2024-08-28T13:47:47Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan dev-aws/account-platform/` as comment.

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: account-platform/kafka-shared
Path: prod-aws/account-platform/
Commit ID: c3873fb948f7407a9d44f5394fa6dbf2de9b5160
Requested At: 2024-08-28T13:47:47Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/account-platform/` as comment.

File: dev-aws/account-platform/topics.tf [hunk 1]
```diff
@@ -94,6 +94,7 @@ resource "kafka_topic" "account_events_v3" {
 resource "kafka_topic" "account_exceptions_events" {
   config = {
     "compression.type" = "snappy"
+    "cleanup.policy"   = "compact"
```
Comment (matiit) [inline dev-aws/account-platform/topics.tf:97]: No need for more compaction settings like retention time etc?

Comment (abramin) [reply dev-aws/account-platform/topics.tf:97]: Not unless we want to change from defaults.

**Review by matiit — APPROVED** [review]

**Comment by uw-infra** [comment]
Received terraform plan request
```
Cluster: prod-aws
Module: account-platform/kafka-shared
Path: prod-aws/account-platform/
Commit ID: 5bf13b3c09711437e641eae4801811357c6d7ab1
Requested At: 2024-08-28T14:24:55Z
```
Do not edit this comment. This message will be updated once the plan run is completed.
To manually trigger plan again please post `@terraform-applier plan prod-aws/account-platform/` as comment.

**Pull request merged**